lsp-haskell.el

```diff
@@ -47,13 +47,15 @@
 
 (defcustom-lsp lsp-haskell-formatting-provider
```

Comment:
could migrate these to `lsp-defcustom` while we're at it. Not sure if it's just a rename or what